### ISSUE TITLE
[PkgEval] MultivariateStats may have a testing issue on Julia 0.4 (2015-01-17)

### ISSUE THREAD

**Issue by IainNZ** [JuliaStats/MultivariateStats.jl#8]
[PackageEvaluator.jl](https://github.com/IainNZ/PackageEvaluator.jl) is a script that runs nightly. It attempts to load all Julia packages and run their tests (if available) on both the stable version of Julia (0.3) and the nightly build of the unstable version (0.4). The results of this script are used to generate a [package listing](http://pkg.julialang.org/) enhanced with testing results.

#### On Julia 0.4
* On **2015-01-16** the testing status was `Tests pass.`
* On **2015-01-17** the testing status changed to `Tests fail, but package loads.`

`Tests pass.` means that PackageEvaluator found the tests for your package, executed them, and they all passed.

`Tests fail, but package loads.` means that PackageEvaluator found the tests for your package, executed them, and they didn't pass. However, trying to load your package with `using` worked.

*This issue was filed because your testing status became worse. No additional issues will be filed if your package remains in this state, and no issue will be filed if it improves. If you'd like to opt-out of these status-change messages, reply to this message saying you'd like to and @IainNZ will add an exception. If you'd like to discuss [PackageEvaluator.jl](https://github.com/IainNZ/PackageEvaluator.jl) please file an issue at the repository. For example, your package may be untestable on the test machine due to a dependency - an exception can be added.*

*Test log*:
```
>>> 'Pkg.add("MultivariateStats")' log
INFO: Installing MultivariateStats v0.1.3
INFO: Package database updated

>>> 'using MultivariateStats' log
Julia Version 0.4.0-dev+2756
Commit 4b20e10 (2015-01-17 03:18 UTC)
Platform Info:
  System: Linux (x86_64-unknown-linux-gnu)
  CPU: Intel(R) Xeon(R) CPU E5-2650 0 @ 2.00GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Sandybridge)
  LAPACK: libopenblas
  LIBM: libopenlibm
  LLVM: libLLVM-3.3

>>> test log
Running tests:
 * lreg.jl
 * whiten.jl

ERROR: LoadError: LoadError: UndefVarError: Triangular not defined
 in cov_whitening at /home/idunning/pkgtest/.julia/v0.4/MultivariateStats/src/whiten.jl:9
 in include at ./boot.jl:248
 in include_from_node1 at ./loading.jl:128
 in anonymous at no file:15
 in include at ./boot.jl:248
 in include_from_node1 at loading.jl:128
 in process_options at ./client.jl:312
 in _start at ./client.jl:396
while loading /home/idunning/pkgtest/.julia/v0.4/MultivariateStats/test/whiten.jl, in expression starting on line 23
while loading /home/idunning/pkgtest/.julia/v0.4/MultivariateStats/test/runtests.jl, in expression starting on line 12
INFO: Testing MultivariateStats
==========================[ ERROR: MultivariateStats ]==========================

failed process: Process(`/home/idunning/julia04/usr/bin/julia --check-bounds=yes --code-coverage=none --color=no /home/idunning/pkgtest/.julia/v0.4/MultivariateStats/test/runtests.jl`, ProcessExited(1)) [1]

================================================================================
INFO: No packages to install, update or remove
ERROR: MultivariateStats had test errors
 in error at error.jl:19
 in test at pkg/entry.jl:717
 in anonymous at pkg/dir.jl:28
 in cd at ./file.jl:20
 in cd at pkg/dir.jl:28
 in test at pkg.jl:68
 in process_options at ./client.jl:234
 in _start at ./client.jl:396


>>> end of log
```

**Comment by maxruby** [comment]
I just tried running the tests on MacOS 10.9.5 and ran into the same ERROR
```julia
julia> include("/Users/maximilianosuster/.julia/MultivariateStats/test/runtests.jl")
Running tests:
 * lreg.jl
 * whiten.jl
ERROR: LoadError: LoadError: UndefVarError: Triangular not defined
```
Will this be fixed any time soon?

**Comment by IainNZ** [comment]
@lindahua is not really available right now, so a pull request from you is probably the best way to get it done. See the Julia issue referred above.

**Comment by maxruby** [comment]
Thanks. I see that the issue of triangular matrix types is not yet resolved? I will try to have a look later anyway.

**Comment by IainNZ** [comment]
It is resolved, its just that the name changed so there needs to be a `if` depending on whether its 0.3 or unstable

**Comment by jiahao** [comment]
Should be fixed in cdd79759d5df56786280437aeded38bf46887bf0
